apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/contacts/ContactListNavView.kt

```diff
@@ -0,0 +1,131 @@
+package chat.simplex.common.views.contacts
+
+import androidx.compose.runtime.*
+import androidx.compose.ui.graphics.Color
+import dev.icerock.moko.resources.compose.painterResource
+import dev.icerock.moko.resources.compose.stringResource
+import chat.simplex.common.model.*
+import chat.simplex.common.platform.*
+import chat.simplex.common.views.chat.*
+import chat.simplex.common.views.chat.item.ItemAction
+import chat.simplex.common.views.chatlist.*
+import chat.simplex.common.views.helpers.*
+import chat.simplex.res.MR
+import kotlinx.coroutines.delay
+
+@Composable
+fun ContactListNavLinkView(chat: Chat, nextChatSelected: State<Boolean>) {
+    val showMenu = remember { mutableStateOf(false) }
+    val loadedChat = remember { mutableStateOf<Chat?>(null) }
+    val rhId = chat.remoteHostId
+    val disabled = chatModel.chatRunning.value == false || chatModel.deletedChats.value.contains(rhId to chat.chatInfo.id)
+    val contactType = getContactType(chat)
+
+    LaunchedEffect(chat.id) {
+        showMenu.value = false
+        delay(500L)
+    }
+
+    LaunchedEffect(contactType) {
+        if (contactType == ContactType.RECENT) {
+            withBGApi {
+                loadedChat.value = chatModel.controller.apiGetChat(rhId, chat.chatInfo.chatType, chat.chatInfo.apiId)
+            }
+        }
+    }
```

Comment:
what does this do? does it preload chat before it's opened, and does so for all "recent" (active) contacts in list? if so, it can be slowing ui quite a lot if there's a lot of active contacts in list, better to only load on opening